src/main/java/org/myrobotlab/service/Log.java

```diff
@@ -283,42 +252,61 @@ public void setRootLogLevel(String level) {
       return;
     } else if ("debug".equalsIgnoreCase(level)) {
       root.setLevel(ch.qos.logback.classic.Level.DEBUG);
-      logLevel = "debug";
     } else if ("info".equalsIgnoreCase(level)) {
       root.setLevel(ch.qos.logback.classic.Level.INFO);
-      logLevel = "info";
     } else if ("warn".equalsIgnoreCase(level)) {
       root.setLevel(ch.qos.logback.classic.Level.WARN);
-      logLevel = "warn";
     } else if ("error".equalsIgnoreCase(level)) {
       root.setLevel(ch.qos.logback.classic.Level.ERROR);
-      logLevel = "error";
     } else {
       log.error("unknown logging level {}", level);
     }
 
     if (!isLogging) {
       root.addAppender(this);
     }
-
+    // getting current level before broadcasting state
+    getLogLevel();
     broadcastState();
   }
 
+  @Override
+  public void start() {
+  }
+
   public void startLogging() {
     // LoggerContext lc = (LoggerContext) LoggerFactory.getILoggerFactory();
     ch.qos.logback.classic.Logger root = (ch.qos.logback.classic.Logger) LoggerFactory.getLogger(Logger.ROOT_LOGGER_NAME);
-    root.setLevel(ch.qos.logback.classic.Level.INFO);
+    // root.setLevel(ch.qos.logback.classic.Level.INFO);
     root.addAppender(this);
     isLogging = true;
-    setRootLogLevel(logLevel);
+    addTask(minIntervalMs, "flush");
   }
 
+  public void startService() {
+    super.startService();
+    startLogging();
+  }
+
+  @Override
+  public void stop() {
+  }
+
+  public String getLog() throws IOException {
+    return FileIO.toString("myrobotlab.log");

```

Comment:
seems like the log file name should be pulled out as a constant, or alternatively, a member variable so you could override it if you wanted.

Reply:
done